__tests__/app.test.js

```diff
@@ -216,3 +213,84 @@ describe("Error handling - /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+// task 7
+
+describe("POST - /api/articles/:article_id/comments", () => {
+  it("201 - should post a comment to the articles table for given article id ", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "This is a comment",
+    };
+    return request(app)
+      .post("/api/articles/9/comments")
+      .send(newComment)
+      .expect(201)
+      .then((response) => {
+        expect(response.body).toBeInstanceOf(Object);
+        expect(response.body.comment).toEqual(
+          expect.objectContaining({
+            comment_id: expect.any(Number),
+            article_id: expect.any(Number),
+            author: expect.any(String),
+            body: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+          })
+        );
+      });
+  });
+});
+
+describe("7. error handling - POST ", () => {
+  it("400 - when article ID is not a number ", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "This is a comment",
+    };
+    return request(app)
+      .post("/api/articles/nonsense/comments")
+      .send(newComment)
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe("Error - Bad Request");
+      });
+  });
+  it("400 - when username or body is missing ", () => {
+    const newComment = {};
+    return request(app)
+      .post("/api/articles/9/comments")
+      .send(newComment)
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe("Error - please enter username or comment");
+      });
+  });
+  it("404 - when username doesnt exist ", () => {
+    const newComment = {
+      username: "emily hassle",
+      body: "This is a comment",
+    };
+    return request(app)
+      .post("/api/articles/9/comments")
+      .send(newComment)
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("Error - Not Found");
+      });
+  });
+  it("404 - when article id is not in database ", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "This is a comment",
+    };
+    return request(app)
+      .post("/api/articles/100000/comments")
+      .send(newComment)
+      .expect(404)
+      .then((res) => {
+        console.log(res);
+        expect(res.body.msg).toBe("Error - Not Found");
+      });
+  });
+});
```

Comment:
Great testing Tom! Could we just add one more please for a 200 that ignores unnecessary properties? I.e. if the newComment has more than the username or body in there, it should just ignore it and still post. 👍

Reply:
hey john! thank you i have done, and the test is failing when I use 200 but passes when I use 201 - is this correct? thanks :)